tests/test_intf_mac.py

```diff
@@ -0,0 +1,258 @@
+        assert src_mac_addr_found == True
+
+        # remove IP from interface
+        self.remove_ip_address("Ethernet8", "10.0.0.4/31")
+
+        # check application database
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:Ethernet8")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 0
+
+    def test_InterfaceChangeMac(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        # assign IP to interface
+        self.add_ip_address("Ethernet12", "12.0.0.4/31")
+
+        # set MAC to interface
+        self.set_mac("Ethernet12", "6C:EC:5A:22:33:44", "12.0.0.4/31")
+
+        # change interface MAC
+        self.set_mac("Ethernet12", "6C:EC:5A:33:44:55", "12.0.0.4/31")
+
+        # check application database
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:Ethernet12")
+        (status, fvs) = tbl.get(tbl.getKeys()[0])
+        assert status == True
+        values = dict(fvs)
+        assert values["mac_addr"] == "6c:ec:5a:33:44:55"
+
+        # check ASIC router interface database
+        src_mac_addr_found = self.find_mac("Ethernet12", "6C:EC:5A:33:44:55")
+        assert src_mac_addr_found == True
+
+        # remove IP from interface
+        self.remove_ip_address("Ethernet12", "12.0.0.4/31")
+
+        # check application database
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:Ethernet12")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 0
+
+class TestLagRouterInterfaceMac(object):
+    def setup_db(self, dvs):
+        self.pdb = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+        self.adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+        self.cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+
+    def create_port_channel(self, dvs, alias):
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL")
+        fvs = swsscommon.FieldValuePairs([("admin_status", "up"),
+                                          ("mtu", "9100")])
+        tbl.set(alias, fvs)
+        time.sleep(1)
+
+    def remove_port_channel(self, dvs, alias):
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL")
+        tbl._del(alias)
+        time.sleep(1)
+
+    def add_port_channel_members(self, dvs, lag, members):
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL_MEMBER")
+        fvs = swsscommon.FieldValuePairs([("NULL", "NULL")])
+        for member in members:
+            tbl.set(lag + "|" + member, fvs)
+            time.sleep(1)
+
+    def remove_port_channel_members(self, dvs, lag, members):
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL_MEMBER")
+        for member in members:
+            tbl._del(lag + "|" + member)
+            time.sleep(1)
+
+    def add_ip_address(self, interface, ip):
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL_INTERFACE")
+        fvs = swsscommon.FieldValuePairs([("NULL", "NULL")])
+        tbl.set(interface + "|" + ip, fvs)
+        time.sleep(1)
+
+    def remove_ip_address(self, interface, ip):
```

Comment:
these functions are already defined in conftest.py file. you could re-use them.

Reply:
@stcheng , the vstest has been updated according to your comments, please review again, thanks.